src/main/java/mate/jdbc/dao/DriverDaoImpl.java

```diff
@@ -0,0 +1,120 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.lib.exception.DataInsertException;
+import mate.jdbc.lib.exception.DataProcessingException;
+import mate.jdbc.model.Driver;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class DriverDaoImpl implements DriverDao {
+
+    @Override
+    public Driver create(Driver driver) {
+        String insertDriverRequest = "INSERT INTO drivers (name,licenseNumber) values(?,?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createDriverStatement =
+                        connection.prepareStatement(insertDriverRequest,
+                             Statement.RETURN_GENERATED_KEYS)) {
+            createDriverStatement.setString(1, driver.getName());
+            createDriverStatement.setString(2,driver.getLicenseNumber());
+            createDriverStatement.executeUpdate();
+            ResultSet generatedKeys = createDriverStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                driver.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataInsertException("Can`t insert driver: "
+                    + driver.getName() + "to DB ", e);
+        }
+        return driver;
+    }
+
+    @Override
+    public Optional<Driver> get(Long id) {
+        String insertDriverRequest =
+                "SELECT * FROM drivers WHERE id = ? and is_deleted = ?;";
```

Comment:
do we need ? in is_deleted

Reply:
yes, we need, user shouldn't be able to get deleted field